R/common.R

```diff
@@ -1003,20 +1003,22 @@ registerData <- function(data) {
    #TODO
 }
 
-checkAnalysisOptions <- function(analysisName, options) {
-   #TODO
-   return(options)
+checkAnalysisOptions <- function(analysisName, options, version) {
+  # TODO when QMLComponents can be linked to jaspBase
+  return(options)
 }
 
-runWrappedAnalysis <- function(analysisName, data, options) {
-   if (jaspResultsCalledFromJasp()) {
-      result <- list("options" = options, "analysis" = analysisName)
-      result <- jsonlite::toJSON(result, auto_unbox = TRUE, digits = NA, null="null", force = TRUE)
-      return(toString(result))
-   } else {
-      registerData(data)
-      options <- checkAnalysisOptions(analysisName, options)
-      return(jaspTools::runAnalysis(analysisName, data, options))
-   }
-}
+runWrappedAnalysis <- function(analysisName, data, options, version) {
+  if (jaspResultsCalledFromJasp()) {
 
+    result <- list("options" = options, "analysis" = analysisName, "version" = version)
+    result <- jsonlite::toJSON(result, auto_unbox = TRUE, digits = NA, null="null", force = TRUE)
+    return(toString(result))
```

Comment:
I don't think `toString` exists as a standalone function (it's a method on jaspObject)?

Reply:
> I don't think `toString` exists as a standalone function (it's a method on jaspObject)?
Hmm... I'm not sure to understand: first jaspObject is in jaspBase, so it should be anyway available. Second the `toString` function does not seem to come from jaspObject anyway. I can just call toString on a jsonlite object in a new session in R Studio without problem.

Reply:
There is a function `base::toString`:

> This is a helper function for `format` to produce a single character string describing an R object.

If you just want to convert the json object to a character, just use `as.character(result)`...